platform/status/src/StatusService.js

```diff
@@ -0,0 +1,92 @@
+/*****************************************************************************
+ * Open MCT Web, Copyright (c) 2014-2015, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+
+define(
+    ['./StatusConstants'],
+    function (StatusConstants) {
+        'use strict';
+
+        var STATUS_PREFIX = StatusConstants.TOPIC_PREFIX;
+
+        /**
+         * The `statusService` maintains information about the current
+         * status of specific domain objects within the system. Status
+         * is represented as string flags which are present when a
+         * domain object possesses that status, and false when it does
+         * not.
+         *
+         * @param {platform/core.Topic} topic the `topic` service, used
+         *        to create/use named listeners.
+         * @constructor
+         * @memberof platform/status
+         */
+        function StatusService(topic) {
+            this.statusTable = {};
```

Comment:
> Is this being done this way because of the possibility of having multiple instances representing the same object? 

It is.

> If so, maybe we need to rethink this. Maybe, for example, we need to consider an object cache so that whenever an object is requested from the object service, the same instance is always returned?

I believe at some point in the past (possibly pre-Angular) that we'd implemented such a thing. I'll have to dig into why this was removed or not kept. Context (what path was used to reach an object) and editing (insulating "real" objects from unsaved changes) both imply different instances of the same object, but those are both handled by wrappers, so that's not a real answer.

A more concrete problem with caches is memory - we do cache domain object _models_ currently, and #193 cites some problems with this. Caching domain objects would exacerbate or at least reproduce the problem (so we'd need some eviction strategy, but I don't know of a reliable way to implement this in JS.) Filed #322 as a follow-up to look at this further